app/controllers/repos_controller.rb

```diff
@@ -92,33 +88,26 @@ def show_releases
           benchmark_runs << latest_benchmark_run
         end
 
-        columns = ChartBuilder.new(benchmark_runs).build_columns do |benchmark_run|
+        chart_builder = ChartBuilder.new(benchmark_runs, benchmark_result_type)
+        chart_builder.build_columns do |benchmark_run|
           environment = YAML.load(benchmark_run.environment)
 
-          if environment.is_a?(Hash)
-            temp = ""
-
-            environment.each do |key, value|
-              temp << "#{key}: #{value}<br>"
-            end
+          version = { version: benchmark_run.initiator.version }
 
-            environment = temp
+          # If there is more information about the environment, we add it to `version` 
+          if environment.is_a?(Hash)
+            version.merge!(environment)
+          else
+            version[:environment] = environment
           end
 
-          "Version: #{benchmark_run.initiator.version}<br> #{environment}"
+          version
         end
-
-        [columns, benchmark_result_type]
       end.compact
     end
 
-    respond_to do |format|
-      format.html do
-        @result_types = fetch_categories
-      end
-
-      format.js
-    end
+    @result_types = fetch_categories if request.format.html?
+    @benchmark_name = params[:result_type]
```

Comment:
If the user input (`params[:result_type]`) is an array, hash or 1GB file upload, is it going to be serialized to JSON and rendered? That sounds like a security vulnerability. 

You'd like to check if the value looks legit. In this case, it's enough to do `@benchmark_name = params[:result_type].to_s` to convert it into string and don't care about the actual type.